src/linear_algebra/projective_space/basic.lean

```diff
@@ -0,0 +1,232 @@
+- `projectivization.mk' K v` where `v : { w : V // w ≠ 0 }`.
+- `projectivization.mk'' H h` where `H : submodule K V` and `h : finrank H = 1`.
+
+## Other definitions
+- For `v : projectivization K V`, `v.submodule` gives the corresponding submodule of `V`.
+- `projectivization.equiv_submodule` is the equivalence between `projectivization K V`
+  and `{ H : submodule K V // finrank H = 1 }`.
+- For `v : projectivization K V`, `v.rep : V` is a representative of `v`.
+
+-/
+
+variables (K V : Type*) [field K] [add_comm_group V] [module K V]
+
+/-- The setoid whose quotient is the projectivization of `V`. -/
+def projectivization_setoid : setoid { v : V // v ≠ 0 } :=
+{ r := λ u v, ∃ a : K, a • (u : V) = v,
+  iseqv := begin
+    refine ⟨λ u, ⟨1, by simp⟩, _, λ u v w ⟨a,ha⟩ ⟨b,hb⟩, ⟨b * a, by simp [ha, hb, mul_smul]⟩⟩,
+    rintros u v ⟨a,ha⟩,
+    use a⁻¹,
+    rw [← ha, ← mul_smul, inv_mul_cancel, one_smul],
+    intros c, apply v.2,
+    simpa [c] using ha.symm,
+  end }
+
+
+/-- The projectivization of the `K`-vector space `V`. -/
+def projectivization := quotient (projectivization_setoid K V)
+
+namespace projectivization
+
+variables {V}
+
+/-- Construct an element of the projectivization from a nonzero vector. -/
+def mk (v : V) (hv : v ≠ 0) : projectivization K V := quotient.mk' ⟨v,hv⟩
+
+/-- A variant of `projectivization.mk` in terms of a subtype. `mk` is preferred. -/
+def mk' (v : { v : V // v ≠ 0 }) : projectivization K V := quotient.mk' v
+
+@[simp] lemma mk'_eq_mk (v : { v : V // v ≠ 0}) :
+  mk' K v = mk K v v.2 :=
+by { dsimp [mk, mk'], congr' 1, simp }
+
+noncomputable instance [nontrivial V] : inhabited (projectivization K V) :=
+let e := exists_ne (0 : V) in ⟨mk K e.some e.some_spec⟩
+
+variable {K}
+
+/-- Choose a representative of `v : projectivization K V` in `V`. -/
+protected noncomputable def rep (v : projectivization K V) : V := v.out'.1
+
+lemma rep_nonzero (v : projectivization K V) : v.rep ≠ 0 := v.out'.2
+
+@[simp]
+lemma mk_rep (v : projectivization K V) :
+  mk K v.rep v.rep_nonzero = v :=
+by { dsimp [mk, projectivization.rep], simp }
+
+open finite_dimensional
+
+/-- Consider an element of the projectivization as a submodule of `V`. -/
+protected def submodule (v : projectivization K V) : submodule K V := K ∙ v.rep
+
+variable (K)
+
+lemma exists_of_mk_eq_mk  (v w : V) (hv : v ≠ 0) (hw : w ≠ 0) (h : mk K v hv = mk K w hw) :
+  ∃ (a : K) (ha : a ≠ 0), a • v = w :=
+begin
+  obtain ⟨a,h⟩ := quotient.exact' h,
+  refine ⟨a,_,h⟩,
+  intros c, refine hw _,
+  simpa [c] using h.symm,
+end
+
+lemma exists_smul_eq_mk_rep
+  (v : V) (hv : v ≠ 0) : ∃ (a : K) (ha : a ≠ 0), a • v = (mk K v hv).rep :=
+exists_of_mk_eq_mk _ _ _ hv (rep_nonzero _) (by simp)
+
+variable {K}
+
+@[simp]
+lemma submodule_mk (v : V) (hv : v ≠ 0) : (mk K v hv).submodule = K ∙ v :=
+begin
+  dsimp only [projectivization.submodule],
+  obtain ⟨a, ha, h⟩ := exists_smul_eq_mk_rep K v hv,
+  rw ← h,
+  apply submodule.span_singleton_smul_eq _ ha,
+end
```

Comment:
This set of definitions makes this `rfl`:
```lean
/-- The setoid whose quotient is the projectivization of `V`. -/
def projectivization_setoid : setoid { v : V // v ≠ 0 } :=
(mul_action.orbit_rel Kˣ V).comap coe

/-- The projectivization of the `K`-vector space `V`. -/
def projectivization := quotient (projectivization_setoid K V)

/-- Consider an element of the projectivization as a submodule of `V`. -/
protected def projectivization.submodule (v : projectivization K V) : submodule K V :=
quotient.lift_on' v (λ v, K ∙ (v : V)) $ begin
  rintro ⟨a, ha⟩ ⟨b, hb⟩ ⟨x, (rfl : x • b = a)⟩,
  exact (submodule.span_singleton_smul_eq _ x.ne_zero),
end
```

Reply:
Done as suggested!